addons/payment/static/src/js/payment_form.js

```diff
@@ -126,6 +126,20 @@ publicWidget.registry.PaymentForm = publicWidget.Widget.extend({
         ev.preventDefault();
 
         const checkedRadio = this.el.querySelector('input[name="o_payment_radio"]:checked');
+        const donationAmountInputEl = this.el.querySelector("#other_amount_value");
+        const otherAmountRadioEl = this.el.querySelector("#other_amount");
+        const isOtherAmountChecked = otherAmountRadioEl ? otherAmountRadioEl.checked : true;
+        if (
+            donationAmountInputEl &&
```

Comment:
Hello @msh-odoo 

https://www.loom.com/share/5b725e7696c84fdc909124cb48e3cfa6?sid=75cf3809-bffa-45cd-ac27-2ceeeef89fca

If we remove the custom amount button from the snippet, the `donationAmountInputEl` will no longer be present. If `donationAmountInputEl` is removed, the process will also halt at `isOtherAmountChecked` since it will not be present either. This is just a precautionary measure from my side to prevent potential issues. However, I leave the final decision up to you.